src/array/primitive/data.rs

```diff
@@ -0,0 +1,32 @@
+use crate::array::{Arrow2Arrow, PrimitiveArray};
+use crate::bitmap::Bitmap;
+use crate::buffer::Buffer;
+use crate::types::NativeType;
+use arrow_data::{ArrayData, ArrayDataBuilder};
+
+impl<T: NativeType> Arrow2Arrow for PrimitiveArray<T> {
+    fn to_data(&self) -> ArrayData {
+        let data_type = self.data_type.clone().into();
+
+        let builder = ArrayDataBuilder::new(data_type)
+            .len(self.len())
+            .buffers(vec![self.values.clone().into()])
+            .nulls(self.validity.as_ref().map(|b| b.clone().into()));
+
+        // Safety: Array is valid
+        unsafe { builder.build_unchecked() }
+    }
+
+    fn from_data(data: &ArrayData) -> Self {
+        let data_type = data.data_type().clone().into();
+
+        let mut values: Buffer<T> = data.buffers()[0].clone().into();
```

Comment:
This is zero copy, right? The `slice` call below is because `arrow2` Array has the offset/len in the `Buffer<T>` itself rather than as separate fields ?